src/googletest.h

```diff
@@ -75,14 +75,21 @@ _END_GOOGLE_NAMESPACE_
 #undef GLOG_EXPORT
 #define GLOG_EXPORT
 
+static vector<string>* temp_directories_list;
+
 static inline string GetTempDir() {
-#ifndef GLOG_OS_WINDOWS
-  return "/tmp";
-#else
-  char tmp[MAX_PATH];
-  GetTempPathA(MAX_PATH, tmp);
-  return tmp;
-#endif
+  if (temp_directories_list == NULL) {
+    temp_directories_list = new vector<string>;
```

Comment:
I'm not happy with this dynamic memory allocation because it will unnecessarily leak memory. The pointer can be easily changed to a normal `std::vector<std::string>` instance.